docs/design-docs/01-kpt-live-stdin.md

```diff
@@ -0,0 +1,242 @@
+# Title
+
+* Author(s): Yuwen Ma <yuwenma@google.com>
+* Contributor(s):
+    - Sean Sullivan <seans@google.com>
+    - Sunil Arora <sunilarora@google.com>
+    - Mortent Torkildsen <mortent@google.com>
+    - Mengqi Yu <mengqiy@google.com>
+    - Natasha Sarkar <natashasarkar@google.com>
+
+* Approver: Sunil Arora <sunilarora@google.com>
+
+## Why
+
+This design is aimed at making `kpt live` be hydration-tool-neutral so that `kpt live`
+applicable use cases can be expanded to `kustomize build`, `helm templates` and raw
+kubernetes manifests.
+
+### Non-goal
+- This doc is not aimed at designing a unified hydration between kpt and kustomize
+- This doc is not aimed at changing the kpt in-place hydration models.
+
+## Design
+
+We propose to make `kpt live` support reading configurations from standard input,
+detach the Inventory from Kptfile and simplify the Inventory-ResourceGroup mechanism
+to use  ResourceGroup only.
+
+### Read standard input resources
+
+We will add a reserved keyword “-” as a special kpt directory. This "-" accepts resources
+from the standard input. For example,  `kustomize build | kpt live apply -`
+
+### Initialize a `ResourceGroup` object
+
+`kpt live init` will create a ResourceGroup CR in resourcegroup.yaml. Only three fields
+should be given to the CR.
+
+- metadata.name: A client-provided valid RFC 1123 DNS subdomain name. Default value has prefix “inventory-” with
+  8 digit numbers. E.g. “inventory-02014045”
+- metadata.namespace: a valid namespace. default to value “default”
+- metadata.labels.cli-utils.sigs.k8s.io/inventory-id: A client-provided valid label.
+  Default value is a UUID e.g. 7c5af957-a3e2-4d68-8c0f-6c1864a66050
+
+### Convert Inventory to ResourceGroup
+
+The client-side Inventory is considered as the identifier of the cluster-side
+ResourceGroup CR. Right now, the Inventory is stored in the Kptfile to guarantee
+the ResourceGroup is unique. This requires Kptfile to exist to use `kpt live apply`.
+
+To split the Inventory from Kptfile to resourcegroup.yaml and convert the Inventory to
+ResourceGroup, `kpt live migrate` should be extended to map the inventory.name,
+inventory.namespace, inventory.inventoryID to  ResourceGroup CR metadata.name,
+metadata.namespace, metadata.labels.cli-utils.sigs.k8s.io/inventory-id correspondingly.
+
+Inventory in Kptfile
+```yaml
+apiVersion: kpt.dev/v1betaX
+kind: Kptfile
+inventory:
+  name: <INVENTORY_NAME>
+  namespace: <INVENTORY_NAMESPACE>
+  inventoryID: <INVENTORY_ID>
+```
+resoucegroup.yaml
+```yaml
+apiVersion: kpt.dev/v1alpha1
+kind: ResourceGroup
+metadata  
+  name: <INVENTORY_NAME>
+  namespace: <INVENTORY_NAMESPACE>
+  labels:
+    cli-utils.sigs.k8s.io/inventory-id: <INVENTORY_ID>
```

Comment:
We should may be include a paragraph explaining how `kpt live apply ` populates this resource with the inventory list etc. and mention never to run `kubectl apply rg.yaml` to the cluster because it will remove the info populated by the `live apply`. And we should also document, how someone can recover if they accidently run `kubectl apply` with rg.yaml ?

Reply:
Defer to @rquitales. We can have the user guide once the feature is complete. You might find more cases want to talk about at that time :)